### ISSUE TITLE
`forum-post-countdown` also starts the countdown on other errors

### ISSUE THREAD

**Issue by mybearworld** [ScratchAddons/ScratchAddons#6926]
### Description

When any error occurs before posting, `forum-post-countdown` starts the 60 second countdown.

### Steps to reproduce

1. Try to post an empty post or `assets.scratch.mit.edu/get_image`

### Environment

Browser: Edge 119
Scratch Addons version: 1.35.1
Operating system: Windows

**Comment by DNin01** [comment]
> I believe this doesn't happen in `auto-follow-topics`

This is what it does:
> https://github.com/ScratchAddons/ScratchAddons/blob/229cde320fd79f3b03268e1e094f1ada79f4471f/addons/auto-follow-topics/userscript.js#L4
> https://github.com/ScratchAddons/ScratchAddons/blob/229cde320fd79f3b03268e1e094f1ada79f4471f/addons/auto-follow-topics/userscript.js#L14

**Comment by mybearworld** [comment]
> > https://github.com/ScratchAddons/ScratchAddons/blob/229cde320fd79f3b03268e1e094f1ada79f4471f/addons/auto-follow-topics/userscript.js#L4
> > 
> > https://github.com/ScratchAddons/ScratchAddons/blob/229cde320fd79f3b03268e1e094f1ada79f4471f/addons/auto-follow-topics/userscript.js#L14

That's odd - other error messages also have `.errorlist` on them.

**Comment by mybearworld** [comment]
https://github.com/ScratchAddons/ScratchAddons/blob/229cde320fd79f3b03268e1e094f1ada79f4471f/addons/forum-post-countdown/userscript.js#L4-L8

These lines would have to mimic `auto-follow-topics`'s behavior instead.

**Comment by DNin01** [comment]
> These lines would have to mimic `auto-follow-topics`'s behavior instead.

I think actually, the problem is that we need to properly check if there is an error message after the page reloads. There doesn't seem to be any code in `forum-post-countdown` that checks for error messages. [`auto-follow-topics` does](https://github.com/ScratchAddons/ScratchAddons/issues/6926#issuecomment-1837262446).

**Comment by DNin01** [comment]
This was also a problem on #4716. Maybe running code on a successful post should be handled by a library.

**Comment by WorldLanguages** [comment]
> Maybe running code on a successful post should be handled by a library.

Yes, it could be moved into a separate js file inside any relevant addon folder, and any other addons could import the utility and use it.

**Comment by DNin01** [comment]
> > Maybe running code on a successful post should be handled by a library.
> 
> Yes, it could be moved into a separate js file inside any relevant addon folder, and any other addons could import the utility and use it.

Would the storage key used to remember the topic that the user just submitted a post on be shared with every addon that uses this library (so there would only be one storage key) or would it be separate for each addon (one storage key for each addon that uses this feature)?

Or could it be handled by an addon API? Or not use session storage at all?

**Comment by DNin01** [comment]
If the library functioned like copy-pasted code, then probably it would be that each addon has its own session storage key for this, which may be the simplest solution.

**Comment by WorldLanguages** [comment]
I don't see any reason to use more than one storage key.

**Comment by DNin01** [comment]
> I don't see any reason to use more than one storage key.

It's just we need to program it in a way where addons don't mess each other up. Our current code removes the topic ID from storage when the post is successful. If we just copy-paste the code and share a single storage key, then what might happen is if one addon sees that the page for the topic ID in storage was reloaded without an error message (which means a successful post) and then removes the topic ID, none of the other addons would know that the user just posted on that topic because the topic ID was removed from storage before they could check.

There are probably ways to avoid this issue while still using a storage key.

**Comment by WorldLanguages** [comment]
Remember that if 2 JavaScript files import the same module, only one instance of the module actually runs. The "was a post just successfully posted" module could run all of its code at the top-level of the file, which would only run once, and expose a (static for the duration of the page) boolean variable/function that addons could read to know whether they should start a countdown or click the follow button.
